keras_cv/layers/object_detection/roi_pool.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv import bounding_box
+
+
+class ROIPooler(tf.keras.layers.Layer):
+    """
+    Pooling feature map of dynamic shape into region of interest (ROI) of fixed shape.
+
+    Mainly used in Region CNN (RCNN) networks. This works for a single-level
+    input feature map.
+
+    This layer splits the feature map into [target_size[0], target_size[1]] areas,
+    and performs max pooling for each area. The area coordinates will be quantized.
+
+    Args:
+        bounding_box_format: a case-insensitive string.
+            For detailed information on the supported format, see the
+            [KerasCV bounding box documentation](https://keras.io/api/keras_cv/bounding_box/formats/).
+        target_size: List or Tuple of 2 integers of the pooled shape
+        image_shape: List of Tuple of 3 integers, or `TensorShape` of the image shape.
+
+    Usage:
+    ```python
+    feature_map = tf.random.normal([2, 16, 16, 512])
+    roi_pooler = ROIPooler("yxyx", [7, 7], [224, 224, 3])
+    rois = tf.constant([[[15., 30., 25., 45.]], [[22., 1., 30., 32.]]])
+    pooled_feature_map = roi_pooler(feature_map, rois)
+    ```
+    """
+
+    def __init__(
+        self,
+        bounding_box_format,
+        target_size,
```

Comment:
naming wise, I think we should consolidate "size" and "shape".

Reply:
Hmm, I'd actually like to have size and shape here because one is a tuple, the other one can be a tuple as well as TensorShape
If we don't like this, I can make it as `target_height`, `target_width`, image_shape

Reply:
I think there was also another discussion about whether kpl should accept "size" vs "height + width", and we should consolidate with that as well.

Reply:
Added a TODO here